### ISSUE TITLE
Release 0.8.0

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1432]
We have done quite a lot of great work recently, so I'd say we should aim for 0.8.0 in following weeks. Below is a list of PRs which are in either mergeable state or very close to it, which I'd like to see merged before the release. Then a couple which needs to be discussed. And a list of issues under 0.8.0 milestone, but I don't think we have to fix all of them before the release.

Key enhancements for 0.8.0:

- optional PyGEOS support and related spatial indexing enhancements
- CRS on GeometryArray level
- Writing to PostGIS + Parquet IO

Blockers:

- [x] Follow-ups on PyGEOS integration:
  - [x] #1404 integrate pygeos.STRtree spatial index
  - [x] #1430
  - [x] #1428
  - [x] #1421
- [x] Parquet IO merge #1180
  - [x] #1409, #1435 add support for to_feather / read_feather 

Other PRs to be included:

- [x] #1385
- [x] #1302
- [x] #1253
- [x] #1123
- [x] #1440
- [ ] #416

PRs for a discussion:
- [x] #1378
- [ ] ~#1376~ (0.9.0)
- [ ] #1324 
- [x] #1290
- [x] #1173

Issues marked for 0.8.0: https://github.com/geopandas/geopandas/milestone/12

- [ ] #498
- [ ] #1179
- [ ] #1345
- [x] #1351
- [x] #1352
- [ ] ~#1230~ (0.9.0)
- [x] #1420

**Comment by jorisvandenbossche** [comment]
Ah, I was just writing exactly the same issue ;) I updated the top post with a few edits from my draft. 

We should indeed put some focus on getting 0.8.0 out.

**Comment by brendan-ward** [comment]
is #1409 indeed a blocker for this release?  Since it would depend on the latest version of pyarrow, it seems like it is more of a nice-to-have than must-have for the short term.  That said, it should be pretty easy once #1180 lands.

**Comment by martinfleis** [comment]
> is #1409 indeed a blocker for this release?

I wouldn't say so, happy to leave it for the next one.

**Comment by adriangb** [comment]
@martinfleis I'd also like to add #1436 regarding spatial indexing. I think it is important to have cohesive error handling now that we have more than one spatial index implementation.

**Comment by martinfleis** [comment]
I'd now leave #1376 (geodesic area and length) for the next release. There is more work than just applying `pyproj`'s implementation if we want to make it robust, so I'd rather focus on the rest now.

**Comment by adriangb** [comment]
I hate to add more things, but if we can at least briefly discuss #1440 before this release, that would be great.

**Comment by jorisvandenbossche** [comment]
We still have a bunch of failures on the pandas master CI, and it should be good to verify that all those cases are things going to be fixed in pandas, and not issues to be fixed in geopandas:

- [ ] clip/overlay failures -> opened https://github.com/pandas-dev/pandas/issues/34741, but we can also change to use tuples in our code anyway
- [ ] setitem failure -> https://github.com/geopandas/geopandas/issues/1372 -> I will try to revive the pandas discussions about that
- [ ] `.astype("string")` failure (due to new added test in pandas) -> this is not critical, but something we can fix in geopandas
- [ ] Failure in `test_numerical_operations` (`idxmin` error type) -> https://github.com/pandas-dev/pandas/issues/32749

**Comment by CNFeffery** [comment]
When could we update geopandas to 0.8.0, I cant't wait to use `feather` to replace `shapefile`! 😁

**Comment by jorisvandenbossche** [comment]
@martinfleis anything remaining you see as a blocker for the release?

#1351 might be good to still do. 
There are a bunch of open visualization PRs that would be nice to include, but on the other hand, it's maybe more important to just do the release (and don't wait too long for the next). 

I have some time tomorrow to look at this.

@CNFeffery I hope next week ;)

**Comment by martinfleis** [comment]
@jorisvandenbossche no real blockers, just that #1351 would be good (I am not able to do it myself in the next couple of weeks I am afraid).

Three (#1253, #1173, #1290) out of four of those plotting PRs were already ready for 0.7.0 release, so I would like to have them in 0.8.0. Skipping two releases, while they seem to be ready to be merged feels unnecessary. I'll resolve merge conflicts.

**Comment by jorisvandenbossche** [comment]
I am thinking to release tomorrow (after updating/merging https://github.com/geopandas/geopandas/pull/1478 and https://github.com/geopandas/geopandas/pull/1468). All good?

**Comment by martinfleis** [comment]
@jorisvandenbossche Thanks for merging/finishing remaining PRs! I think we're ready to go. Tag me in #1468 once ready for review.

**Comment by jorisvandenbossche** [comment]
0.8.0 is tagged, released on github (https://github.com/geopandas/geopandas/releases/tag/v0.8.0), and uploaded to PyPI. Will announce once the conda-forge packages are ready.

**Comment by kylebarron** [comment]
It looks like the docs website is still on 0.7.0?

**Comment by martinfleis** [comment]
@kylebarron never-ending story #290. Thanks for the notice, I'll update it.

**Comment by kylebarron** [comment]
Oh sorry 😂 I didn't know there were two. I think the first result on Google is the older website

**Comment by martinfleis** [comment]
We are trying to avoid this duplication, so hopefully, this is the last release with this issue and with two versions of the documentation.

**Comment by jorisvandenbossche** [comment]
Hmm .. until this is fixed, I should put it on the checklist for doing a release -> https://github.com/geopandas/geopandas/wiki/Release-checklist

**Comment by jorisvandenbossche** [comment]
Is there appetite for a small bug-fix release? (maybe #1486, #1497, #1503, not sure if there are others, or those 3 are worth it)

**Comment by martinfleis** [comment]
We can do one. Also because #1485 is not part of 0.8.0 docs on geopandas.org.
